ideascube/configuration/__init__.py

```diff
@@ -40,15 +44,46 @@ def get_config(namespace, key):
         return default_value
 
     if not isinstance(config.value, expected_type):
-        logger.error(
-            'The stored value for %s is of type %s instead of %s. This should'
-            ' never have happened.'
-            % (config, type(config.value), expected_type))
+        logger.error(IMPOSSIBLE_MESSAGE % (
+            config, type(config.value), expected_type))
         return default_value
 
     return config.value
 
 
+def reset_config(namespace, key):
+    """Reset a configuration option to its default value
+
+    Args:
+        namespace (str): The configuration namespace.
+        key (str): The configuration key inside the namespace.
+
+    Raises:
+        ideascube.configuration.exceptions.NoSuchConfigurationNamespaceError:
+            The requested namespace does not exist.
+        ideascube.configuration.exceptions.NoSuchConfigurationKeyError:
+            The requested key does not exist in the requested namespace.
+    """
+    from .models import Configuration
+    from .registry import get_expected_type
+
+    expected_type = get_expected_type(namespace, key)
+
+    try:
+        config = Configuration.objects.get(namespace=namespace, key=key)
+
+    except Configuration.DoesNotExist:
+        # The configuration had never been modified, nothing to reset
+        pass
+
+    else:
+        if not isinstance(config.value, expected_type):
+            logger.error(IMPOSSIBLE_MESSAGE % (
+                config, type(config.value), expected_type))
+
+        config.delete()
+
```

Comment:
First of all, I dislike the argument of authority. It does a disservice to the rest of your argument.

As for the actual issue, I'm used to the `try/except/else` clause and quite like it. There are even cases where I would argue it is better and more understandable than the alternative.

But this is not such a case. This is a case where both possibilities are just as understandable as the other, the only difference being that your suggestion removes some indentation.

I do dislike unnecessary indentation, so I've changed the code to the pattern you suggested in the latest push to this branch.